azurerm/internal/services/peering/tests/resource_arm_peering_asn_test.go

```diff
@@ -0,0 +1,219 @@
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPeerAsnExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMPeerAsn_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMPeerAsnExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Peering.PeerAsnsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Peer Asn not found: %s", resourceName)
+		}
+
+		id, err := parse.PeerAsnID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.Name); err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Peer Asn %q does not exist", id.Name)
+			}
+			return fmt.Errorf("failed to get on Peering.PeerAsns: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMPeerAsnDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Peering.PeerAsnsClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_peer_asn" {
+			continue
+		}
+
+		id, err := parse.PeerAsnID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if resp, err := client.Get(ctx, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("failed to get on Peering.PeerAsns: %+v", err)
+			}
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccAzureRMPeerAsn_basic(data acceptance.TestData) string {
+	template := testAccAzureRMPeerAsn_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_peer_asn" "test" {
+  name = "acctestASN_%d"
+  asn  = 123
+  contact {
+    role  = "Noc"
+    email = "email@test.com"
+  }
+  peer_name = "acctest-peer"
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMPeerAsn_complete(data acceptance.TestData) string {
+	template := testAccAzureRMPeerAsn_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_peer_asn" "test" {
+  name = "acctestASN_%d"
+  asn  = 123
+  contact {
+    role  = "Noc"
+    email = "email@test.com"
+    phone = 12345
+  }
+  contact {
+    role  = "Service"
+    email = "email@test.com"
+    phone = 12345
+  }
+  peer_name = "acctest-peer2"
+}
+`, template, data.RandomInteger)
+}
+
+func testAccAzureRMPeerAsn_requiresImport(data acceptance.TestData) string {
+	template := testAccAzureRMPeerAsn_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_peer_asn" "import" {
+  name = azurerm_peer_asn.test.name
+  asn  = azurerm_peer_asn.test.asn
+  contact {
+    role  = "Noc"
+    email = "email@test.com"
+  }
+  peer_name = azurerm_peer_asn.test.peer_name
+}
+`, template)
+}
+
+func testAccAzureRMPeerAsn_template(data acceptance.TestData) string {
+	return `provider "azurerm" {
+  features {}
+}
```

Comment:
I scan through the current codebase and see only a few tests have a non-empty provider block, which is mainly to test for some feature of itself or what it depends on.
In this case, since I have no such requirement to tune provider block among test cases, do you feel it is OK to reuse that block?